rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java

```diff
@@ -168,7 +168,7 @@ static List<Path> dependenciesFromResources(ExecutionContext ctx, String... arti
                         .forName("org.openrewrite.java.Java25Parser")
                         .getDeclaredMethod("builder")
                         .invoke(null);
-            } catch (Exception e) {
+            } catch (Throwable e) {
```

Comment:
This now also catches `UnsupportedClassVersionError`, meaning while'll continue down to try other parsers as well.